src/file.js

```diff
@@ -12,6 +12,58 @@ var utils = require('./utils');
 
 var file = module.exports;
 
+file.copyFile = function (src, dst, opts, cb) {
+	/* Wrapper for copyFile function with code to handle it not being present. If/When nodejs 8.5 or greater is a baseline requirement/assumption,
+	   this can be pruned and wholesale replace in all calls with fs.copyFile. */
+
+	function noop() {}
+
+	function copyHelper(err) {
+		var is;
+		var os;
+
+		if (!err && !(opts.replace || opts.overwrite)) {
+			return cb(new Error('File ' + dst + ' exists.'));
+		}
+
+		fs.stat(src, function (err, stat) {
+			if (err) {
+				return cb(err);
+			}
+
+			is = fs.createReadStream(src);
+			os = fs.createWriteStream(dst);
+
+			is.pipe(os);
+			os.on('close', function (err) {
+				if (err) {
+					return cb(err);
+				}
+
+				fs.utimes(dst, stat.atime, stat.mtime, cb);
+			});
+		});
+	}
+
+	/* Test for availability of fs.CopyFile */
+	const semver = require('semver');
+
+	if (semver.gte(process.version, '8.5.0')) {
```

Comment:
Just use `typeof fs.copyFile !== 'function'` instead. No need for semver. It's safer to check if the function is there.

Reply:
I hadn't thought of it that way. Good point. I'll make that change.